src/network/node.rs

```diff
@@ -130,7 +130,7 @@ impl Node {
         }
     }
 
-    pub async fn run(&mut self) {
+    pub async fn run(mut self) {
```

Comment:
this is an important change since when using `async_std::task::spawn`, you should give the function ownership of the item it is consuming. this is a bit over my head as for the details but you can probably find more about this online